DAOS-10595 build: Update DPDK to v21.11.1 to align with SPDK v22.01.1

tanabarr opened the pull request:
Update DPDK RPM version to 21.11.1 to align with SPDK 22.01.1.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

tanabarr committed:
DAOS-10595 build: Update DPDK to v21.11.1

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

tanabarr committed:
add codeowners file

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

brianjmurrell reviewed (changes requested):
Per the build failure, you need to update the `Jenkinsfile`.  Also, could you rebase your `packaging/` subdir on `bmurrell/rpmlint-support`?  Just looking for a few more tests of that before landing it to master.  Much appreciated.

tanabarr committed:
update Jenkinsfile to el8

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

tanabarr committed:
use bmurrell/rpmlint-support packaging branch and supply -n dpdk-stable to %setup as the extracted directory name

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

tanabarr committed:
use bmurrell/rpmlint-support packaging branch and add patch

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

tanabarr committed:
package dumpcap binary

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

tanabarr committed:
start patch directives from 0 to make them apply correctly

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

tanabarr committed:
change %setup to %autosetup in %prep

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

tanabarr committed:
exclude examples from devel package

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

tanabarr committed:
remove missing binaries from package files

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

tanabarr committed:
remove empty tools package

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

tanabarr committed:
make configure consistent on debian

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

tanabarr committed:
Skip Ubuntu

PR-repos: spdk@PR-47:7
PR-repos: daos@PR-9063

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

tanabarr committed:
add rpmlintrc

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

tanabarr committed:
add ubuntu back to Jenkinsfile

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

brianjmurrell reviewed (changes requested):
Looks like you need to deal with the `debian/` equivalent of https://github.com/daos-stack/dpdk/pull/22/files#diff-ef707faf4a45dab41b6870e7e20cf390bad55fa4596b42532483fa11cd09f745L161-L163 by removing the now missing files in `debian/*.install` files:

```
[2022-05-20T16:41:54.428Z] dh_install: warning: Cannot find (any matches for) "usr/bin/dpdk-devbind.py" (tried in ., debian/tmp)
[2022-05-20T16:41:54.428Z] 
[2022-05-20T16:41:54.428Z] dh_install: warning: dpdk missing files: usr/bin/dpdk-devbind.py
[2022-05-20T16:41:54.428Z] dh_install: warning: Cannot find (any matches for) "usr/bin/dpdk-hugepages.py" (tried in ., debian/tmp)
[2022-05-20T16:41:54.428Z] 
[2022-05-20T16:41:54.428Z] dh_install: warning: dpdk missing files: usr/bin/dpdk-hugepages.py
[2022-05-20T16:41:54.428Z] dh_install: warning: Cannot find (any matches for) "usr/bin/dpdk-pmdinfo.py" (tried in ., debian/tmp)
[2022-05-20T16:41:54.428Z] 
[2022-05-20T16:41:54.428Z] dh_install: warning: dpdk missing files: usr/bin/dpdk-pmdinfo.py
[2022-05-20T16:41:54.428Z] dh_install: warning: Cannot find (any matches for) "app/test/autotest.py" (tried in ., debian/tmp)
[2022-05-20T16:41:54.428Z] 
[2022-05-20T16:41:54.428Z] dh_install: warning: dpdk-dev missing files: app/test/autotest.py
[2022-05-20T16:41:54.428Z] dh_install: warning: Cannot find (any matches for) "app/test/autotest_data.py" (tried in ., debian/tmp)
[2022-05-20T16:41:54.428Z] 
[2022-05-20T16:41:54.428Z] dh_install: warning: dpdk-dev missing files: app/test/autotest_data.py
[2022-05-20T16:41:54.428Z] dh_install: warning: Cannot find (any matches for) "app/test/autotest_runner.py" (tried in ., debian/tmp)
[2022-05-20T16:41:54.428Z] 
[2022-05-20T16:41:54.428Z] dh_install: warning: dpdk-dev missing files: app/test/autotest_runner.py
[2022-05-20T16:41:54.428Z] dh_install: warning: Cannot find (any matches for) "app/test/autotest_test_funcs.py" (tried in ., debian/tmp)
[2022-05-20T16:41:54.428Z] 
[2022-05-20T16:41:54.428Z] dh_install: warning: dpdk-dev missing files: app/test/autotest_test_funcs.py
[2022-05-20T16:41:54.428Z] dh_install: warning: Cannot find (any matches for) "usr/bin/dpdk-telemetry.py" (tried in ., debian/tmp)
[2022-05-20T16:41:54.428Z] 
[2022-05-20T16:41:54.428Z] dh_install: warning: dpdk-dev missing files: usr/bin/dpdk-telemetry.py
[2022-05-20T16:41:54.686Z] dh_install: error: missing files, aborting
```

tanabarr committed:
attempt to fix deb packaging build failures

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

tanabarr committed:
disabling ubuntu build temporarily

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

tanabarr committed:
update packaging

PR-repos: spdk@PR-47:17
DAOS-test-branch-master: tanabarr/spdk-update-v22_01_1
DAOS-test-branch-release/2.2: tanabarr/spdk-pin-ltv22-rel2_2
DAOS-test-branch-release/2.0: tanabarr/spdk-pin-ltv22-rel2_0

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

brianjmurrell commented:
~~You don't need `PR-repos: spdk@PR-47:14` in this PR as there is no dependency on spdk in dpdk, correct?~~
You do need them for the test build, but you need to use the distro specific ones like I did in https://github.com/daos-stack/dpdk/pull/23/commits/3754cf65c6afbad110b87bec7cd9bb2ff8a62d1d

This will not work with `PR-repos: daos@PR-9063:16`.  I suspect you are actually looking for:

```
DAOS-test-branch-master: tanabarr/spdk-update-v22_01_1
DAOS-test-branch-release/2.2: bmurrell/packaging-nobest-2.2
DAOS-test-branch-release/2.0: bmurrell/packaging-nobest-2.0
```

At least until daos-stack/daos#9211 and daos-stack/daos#9234 land.

brianjmurrell committed:
Fix debian packaging (#23)

* Fix debian packaging
* Add missing lib packages
* Build debuginfo packages on Leap
* Add more linting exceptions
* Fix expressions

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

brianjmurrell committed:
Fix pragmas

DAOS-test-branch-master: tanabarr/spdk-update-v22_01_1
DAOS-test-branch-release/2.2: bmurrell/packaging-nobest-2.2
DAOS-test-branch-release/2.0: bmurrell/packaging-nobest-2.0
PR-repos: spdk@PR-47

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

brianjmurrell committed:
Fix pragmas

DAOS-test-branch-master: tanabarr/spdk-update-v22_01_1
DAOS-test-branch-release/2.2: bmurrell/packaging-nobest-2.2
DAOS-test-branch-release/2.0: bmurrell/packaging-nobest-2.0

# NOTE: The following default set of pragmas will result in no build
#       or test.
#
#       Please edit the list of pragmas to produce the minimal amount of build
#       and test to prove your PR works.
#
#       Or if you have already done the above, and your PR is ready for it's
#       final for-landing run, remove all of the following pragmas.
#
#       More detail can be found at
#       https://wiki.hpdd.intel.com/display/CI/Commit+Pragmas.
Skip-checkpatch: true
Skip-build: true
# Or you can skip specific PR build stages:
#Skip-build-ubuntu20-rpm: true
#Skip-build-el7-rpm: true
#Skip-build-leap15-rpm: true
#Skip-build-centos7-gcc: true
#Skip-build-centos7-gcc-debug: true
#Skip-build-centos7-gcc-release: true
#Skip-build-leap15-gcc: true
#Skip-build-leap15-icc: true
#Skip-build-ubuntu-clang: true

# And/or you could just enable quick-build below which reduces build tims considerably
Quick-build: true

Skip-unit-tests: true
# Or you could choose which Unit Test stages to skip below
#Skip-nlt: true
#Skip-unit-test: true
#Skip-unit-test-memcheck: true
Skip-test: true

# Or you can choose which test stages to skip below
#Skip-coverity-test: true
# Skip all Functional test stages
#Skip-func-test: true
# Skip Functional test stages on VMs
#Skip-func-test-vm: true
# Skip all Functional test stages on hardware
#Skip-func-hw-test: true
# or just specific hardware test stages
#Skip-func-hw-test-small: true
#Skip-func-hw-test-medium: true
#Skip-func-hw-test-large: true
#Skip-scan-centos-rpms: true
#Skip-test-centos-rpms: true

# If you do run any of the above Functional test stages, please use the following to
# limit your tests run to the tests/features you are working on
# You can add the tags for your tests to the appropriate pragmas below:
#Test-tag: -hw
#Test-tag-hw-small: hw,msall
#Test-tag-hw-medium: hw,medium,ib2
#Test-tag-hw-large: hw,large

# if you want to test your PR with a PR of a component, you can speify that PR:
#PR-repos: project@PR-number[:build_num]

# if you want to disable using RPMs in the Functional tests:
#RPM-test: false

# Or if you want to test with daos RPMs that are already built (i.e. in another PR or in the repo):
#RPM-test-version: version-release

# This should be figured out automatically, but you can force it with
#Doc-only: true

# If you are ready to push a full CI build and test you can uncomment
# the below pragma:
#Full-CI-run: true
Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

brianjmurrell commented on debian/rules:
```diff
@@ -85,7 +85,8 @@ override_dh_auto_configure:
 		-Dinclude_subdir_arch=../$(DEB_HOST_MULTIARCH)/dpdk \
-		-Dmachine=$(RTE_MACHINE)
+		-Ddisable_drivers=compress/isal,compress/mlx5,net/mlx4,net/mlx5,vdpa/mlx5,common/mlx5,regex/mlx5,raw/ioat \
+		-Dmachine=generic
```
Why is this being changed from `default` (line 25) to `generic` here, and why is this change not being made on line 25 to preserve the functionality of the `RTE_MACHINE` override?

tanabarr replied:
default no longer seemed to work and generic matches what spdk builds  dpdk with. I would prefer to land like this and make the suggested changes regarding RTE_MACHINE in a subsequent PR so we can get https://github.com/daos-stack/daos/pull/9063 which is ready to land onto master so we can start working with updated SPDK. does that sound okay?

brianjmurrell replied:
Except that we are producing a release that we have knowingly introduced a regression in functionality.

I don't see a reason to need to re-run daos-stack/daos#9063 just because this is fixed.

brianjmurrell replied:
We also need to land the DAOS-test-branch-*: *nobest* PRs first or he test phase of landing this will fail.

tanabarr replied:
@brianjmurrell great, I will repush with line 25 changes and update the changelog dates on this PR.
Is there anything that's stopping the nobest PRs from landing or can we get those landed before I repush?

brianjmurrell commented on dpdk.spec:
```diff
@@ -193,6 +171,10 @@ CFLAGS="$(echo %{optflags} -fcommon)" \
 %endif
 
 %changelog
+* Tue May 17 2022 Tom Nabarro <tom.nabarro@intel.com> - 0:21.11.1-1
```
Could update the date here in any new push.

brianjmurrell commented on debian/changelog:
```diff
@@ -1,3 +1,12 @@
+  * Update DPDK to 21.11.1 to align with the SPDK 22.01.1 release.
+  * Add additional patches to align with commit pinned by SPDK 22.01.1.
+
+ -- Tom Nabarro <tom.nabarro@intel.com>  Tue, 17 May 2022 10:50:11 -0400
```
Could update the date here in any new push.

tanabarr committed:
Address review comments

DAOS-test-branch-master: tanabarr/spdk-update-v22_01_1
DAOS-test-branch-release/2.2: bmurrell/packaging-nobest-2.2
DAOS-test-branch-release/2.0: bmurrell/packaging-nobest-2.0
PR-repos: spdk@PR-47

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>
